buildlib/azure-pipelines.yml

```diff
@@ -10,33 +10,65 @@ resources:
     - container: centos7
       image: ucfconsort.azurecr.io/ucx/centos7:1
       endpoint: ucfconsort_registry
+    - container: fedora
+      image: ucfconsort.azurecr.io/ucx/fedora:30
+      endpoint: ucfconsort_registry
 
 stages:
   - stage: Build
     jobs:
-      - job: Compile
-        displayName: Compile Tests
-        pool:
-          vmImage: 'Ubuntu-16.04'
+      - job: latest_cc
+        displayName: Latest CCs and CppCheck
+        container: fedora
         steps:
-          - bash: |
-              ./autogen.sh
+          - bash: ./autogen.sh
             displayName: Setup autotools
 
           - bash: |
-              mkdir build && cd build
-              ../configure --disable-numa
-            displayName: Configure
+              mkdir build-gcc && cd build-gcc
+              # cscppc wraps gcc to use its output for cppcheck
+              export PATH="`cscppc --print-path-to-wrap`:$PATH"
+              ../contrib/configure-release
+              make -j`nproc` V=1 2>&1 | tee cc.log
+            displayName: GCC
 
           - bash: |
-               cd build
-               gcc -v
-               make -s -j `nproc`
-            displayName: Build gcc 5.4
+              cd build-gcc
+              cppcheck --version
+
+              cppcheck_err="cppcheck.err"
+              # use cs* tools to keep only UCX related issues
+              cslinker --quiet cc.log \
+                | csgrep --mode=json --path $(dirname $PWD) --strip-path-prefix $(dirname $PWD) \
+                | csgrep --mode=json --invert-match --path 'conftest.c' \
+                | csgrep --mode=grep --invert-match --event "internal warning" --prune-events=1 \
+                > $cppcheck_err
+
+                if [ -s $cppcheck_err ]; then
+                  echo "CppCheck found errors:"
+                  cat $f
+                  fals
+                else
+                  echo "No errors reported by cppcheck"
+                  true
+                fi
+
+            displayName: CppCheck
+
+          - bash: |
+              mkdir build-clang && cd build-clang
+              ../contrib/configure-release CC=clang CXX=clang++
+            displayName: Configure for Clang
+
+          - bash: |
+              cd build-clang
+              clang --version
+              make -j`nproc` V=1
```

Comment:
IMHO you should not use V=1 for CI. This just makes it much harder to find any error/warning output. The CI is reproducible so if someone does need to investigate command line arguments they can just run the command again.

Reply:
I'd rather keep it so CI would provide as much info as it can in a single run, not bothering developers to change anything and re-run.

Reply:
Why would any developer care about the command line options to gcc in CI? That isn't how this stuff really gets used.. The most important thing is to make it easy to find the errors to fix them.